### PR TITLE
[BC break] Implemented non-interactive operation of `dependencies:link` command

### PR DESCRIPTION
**Pull request by alongosz**
| :ticket: Issue | n/a |
|----------------|-----------|

#### Description:
When I have a list of many pull requests to create `dependencies.json`, it's not very convenient to input them interactively one by one, but rather pass that list as arguments. Moreover ATM running `dependencies:link` command in a non-interactive way (`-n` option, built-in by SF) is not possible.

I'm suggesting here an alternative input (not removing current interaction, it's good!). If you pass PR links as arguments, e.g.:

```sh
php bin/ci dependencies:link https://gitbub.com/ibexa/core/pull/1 https://gitbub.com/ibexa/admin-ui/pull/2
```
the interactive prompt gets skipped.

**This unfortunately led to a BC break**, because till now, custom GitHub token was an argument. To make the change possible, I've changed it to be an option (`--token=`). GitHub token is usually not passed anyway, but read from composer configuration. But maybe there's a use case for which this was added and breaking it would create more issues than it solves? Feedback appreciated.

At the same time I've moved token fetching from `interact` to `initialize` method, as this is not a part of interactive prompt and this way it works for both cases.

#### For QA:

- Test new use case
- See the old use case is working
- Test with custom `--token` option

### CODE REVIEW

**Review by tomaszszopinski — APPROVED** [review]
👍

**Review by mnocon — APPROVED** [review]
This change definitely makes sense, I really like it 😄 Thank you!

File: src/Command/LinkDependenciesCommand.php [hunk 1]
```diff
@@ -73,7 +72,8 @@ protected function configure(): void
         $this
             ->setName('dependencies:link')
             ->setDescription(sprintf('Outputs a %s file that contains data about related Pull Requests', self::DEPENDENCIES_FILE))
-            ->addArgument('token', InputArgument::OPTIONAL, 'GitHub OAuth token')
+            ->addOption('token', 't', InputOption::VALUE_OPTIONAL, 'GitHub OAuth token')
```
Comment (adamwojs) [inline src/Command/LinkDependenciesCommand.php:75]:
```suggestion
            ->addOption('token', 't', InputOption::VALUE_REQUIRED, 'GitHub OAuth token')
```

What's the point of providing only `-t` in cmd?

Comment (alongosz) [reply src/Command/LinkDependenciesCommand.php:75]: > What's the point of providing only `-t` in cmd?

@adamwojs I'm not sure, token was there before, I just moved it to the options. We'd need to ask the author of this.

Comment (mnocon) [reply src/Command/LinkDependenciesCommand.php:75]: Let's just say I'm not too attached to this shortcut, we can remove it 😄

**Review by micszo — APPROVED** [review]
I'm so late to the party here. 😄 Thanks!

**Pull request merged**